pkg/skaffold/docker/syncrules.go

```diff
@@ -45,16 +46,29 @@ func SyncRules(workspace string, dockerfilePath string, buildArgs map[string]*st
 
 	for _, copyCommand := range copyCommands {
 		for _, copySrc := range copyCommand.srcs {
-			fi, err := os.Stat(filepath.Join(workspace, copySrc))
-
 			var src, strip string
-			if err != nil || fi.Mode().IsRegular() {
+
+			if strings.Contains(copySrc, "*") {
```

Comment:
Do you think the current code doesn't handle this properly?

Reply:
After thinking about it, I think that my example above works fine.

---

However, I think I found another problem (most people would say it's a docker problem, though.. [1]).

Given this copy command:
```Dockerfile
COPY * ./
```
and the files `nested/deep/foo` and `bar`, the sync rules will become `{Src: "*", Dest: "./"}`. So this will put `nested/deep/foo -> /nested/deep/foo` and `bar -> /bar`. However, Docker flattens the first level of directories on the source side, so the files should actually end up in `nested/deep/foo -> /deep/foo` (bad) and `bar -> /bar` (good).

[1] so many angry users... `https://github.com/moby/moby/issues/15858`

Reply:
Any idea what we should do?

Reply:
What about always stripping the first directory level when we encounter a `*`? That works if the file is in the workdir root (no change), and when it's in a possibly nested directory. Or am I missing something?

Reply:
..oh no! Doesn't this apply to the non-wildcard case as well, if the source is a directory?

Reply:
I'll try a few things tomorrow

Reply:
That's really a mess. I couldn't find a good way to support all the use cases. I'll try again tomorrow. Either I missed something or we can change the way SyncRules work